scripts/009_update_landiq.R

```diff
@@ -0,0 +1,317 @@
+)  |> 
+  terra::project("epsg:3310")
+
+dwr_2016 <- terra::vect(
+  file.path(raw_data_dir, "cadwr_land_use", "LandIQ_shapefiles", "i15_Crop_Mapping_2016_SHP", "i15_Crop_Mapping_2016.shp")
+) |>
+  terra::project("epsg:3310")
+
+## Get overlapping points
+dwr_x <- terra::intersect(dwr_2018, dwr_2016)
+
+## Goal is to reconcile dwr_2016 Unique_ID with dwr_2018 UniqueID
+## join them by geometry, creating id_2016 and id_2018
+## find where UniqueID has changed (or not)
+## where 2016 is missing from 2018, find nearest polygon and calculate distance 
+dwr_merged <- terra::intersect(dwr_2018, dwr_2016) |>
+  # Project to WGS84 (decimal degrees) before converting to dataframe
+  terra::project("epsg:4326") |>
+  as.data.frame(geom = "xy") |> 
+  select(contains("id"), x, y) |>
+  rename(lon = x, lat = y) |>
+  mutate(
+    lon = round(lon, 5),
+    lat = round(lat, 5),
+    match_type = "intersect"
+  )
+
+unmatched <- anti_join(design_points, design_points_ids, by = "site_id") |>
+  distinct()
+# Find nearest polygons for unmatched points
+# Then append to the design_points_ids
+if (nrow(unmatched) > 0) {
+  unmatched_vect <- terra::vect(
+    unmatched, geom = c("lon", "lat"), crs = "epsg:4326"
+  ) |> terra::project("epsg:3310")
+  
+  # Calculate distance matrix and find nearest polygons
+  nearest_data <- tibble(
+    site_id = unmatched$site_id,
+    point_index = seq_len(nrow(unmatched))
+  ) |>
+    mutate(
+      nearest_info = purrr::map(point_index, function(i) {
+        distances <- terra::distance(unmatched_vect[i], dwr_2018)
+        min_idx <- which.min(distances)
+        nearest_poly <- dwr_2018[min_idx]
+        # Get the point coordinates in decimal degrees
+        point_wgs84 <- terra::project(unmatched_vect[i], "epsg:4326")
+        coords <- terra::crds(point_wgs84)
+        tibble(
+          UniqueID = nearest_poly$UniqueID,
+          distance = min(distances),
+          lon = round(coords[1], 5),
+          lat = round(coords[2], 5)
+        )
+      })
+    ) |>
+    tidyr::unnest(nearest_info) |>
+    select(-point_index) |>
+    mutate(match_type = "nearest")
+
+
+  PEcAn.logger::logger.info("Design points joined to nearest polygon:")
+  nearest_data |>
+    knitr::kable(digits = 5) 
+  
+  # Combine intersected points with nearest points
+  design_points_ids_updated <- bind_rows(design_points_ids, nearest_data)
+}
+  
+
+write.csv(design_points_ids_updated |>
+  select(UniqueID, site_id, lat, lon) |>
+  left_join(design_points |> 
+      select(site_id, pft),
+    by = "site_id") |>
+  arrange(pft, lat, lon),
+  here::here("data/design_points.csv"),
+  row.names = FALSE
+)
+
+### Data Summaries
+crops_all |>
+  filter(year == 2018) |>
+  group_by(COUNTY) |>
+  summarize(
+    n_fields = n_distinct(UniqueID),
+    n_classes = n_distinct(CLASS),
+    .groups = "drop"
+  ) |>
+  arrange(desc(n_fields)) |>
+  print(n = 58)
```

Comment:
Why 58?

Reply:
Number of counties in California

Reply:
```suggestion
  print(n = 58) # one per county in CA
```

Sometimes I like to turn magic numbers like this into assertions  like `stopifnot(nrow(df) == 58) # expect one per county`, but prob not that important here